domains/pallets/domain-registry/src/lib.rs

```diff
@@ -179,6 +179,15 @@ mod pallet {
     pub(super) type DomainTotalStakeWeight<T: Config> =
         StorageMap<_, Twox64Concat, DomainId, T::StakeWeight, OptionQuery>;
 
+    /// Map of primary block number to primary block hash.
+    ///
+    /// NOTE: The oldest block hash will be pruned once the oldest receipt is pruned. However, if the
+    /// execution chain stalls, i.e., no receipts are included in the primary chain for a long time,
+    /// this mapping will grow indefinitely.
```

Comment:
But this pallet is included in execution chain itself. Comment was copy-pasted and likely needs to be adjusted.

Reply:
Updated docs and _fixed_ it since it must be domain-aware as well, otherwise in case some core domain stalls but the other domains still moves forward as normal, this state become corrupted.